### ISSUE TITLE
isStartLinearAdModeExpected() check

### ISSUE THREAD

**Issue by mysuf** [videojs/videojs-contrib-ads#363]
Hi. 

Is there public method/var to check that startLinearAdMode is expected and is safe to call skipLinearAdMode? I am trying to handle adserror, but I need to know if I am in adtimeout window.

I tried:

```
if (ads.inAdMode() && !ads.inAdBreak()) {
  // called
  ads.skipLinearAdMode();
}
```

But I've got:

> VIDEOJS: ADS: ContentPlayback -> Midroll
> VIDEOJS: ADS: Starting ad break
> VIDEOJS: Ad error: There was an error playing the video ad.
> VIDEOJS: ADS: Ending ad break
> VIDEOJS: WARN: Unexpected skipLinearAdMode invocation (State via Midroll)
> VIDEOJS: ADS: Midroll -> ContentPlayback

Thanks.

**Comment by incompl** [comment]
adserror is deprecated because of stuff like this. More info here: https://github.com/videojs/videojs-contrib-ads#deprecated-events

I expect it will be easier to get your code to work if you don't trigger adserror.

**Comment by mysuf** [comment]
It seems like missunderstanding. I asked just because I would like to avoid adserror. Thats why I need to know internal state of contrib-ads described above. Thank you.

**Comment by incompl** [comment]
In your console log, what is causing `VIDEOJS: Ad error: There was an error playing the video ad.`? I assumed that meant you were triggering adserror before skipLinearAdMode, which would cause the issue you describe. Maybe I'm misinterpreting the log.

**Comment by mysuf** [comment]
Oh. Sorry. Thats console.warn from my integration (not contrib-ads's error).

**Comment by mysuf** [comment]
This is how I handle it atm: https://github.com/mysuf/videojs-ima/blob/master/src/ima-player.js#L346 It is ugly..

**Comment by incompl** [comment]
I just realized this is a midroll case: would it work to instead call endLinearAdMode?

We have considered adding isAdLoading and isContentResuming, which might also help.

**Comment by mysuf** [comment]
I dont know IMA SDK internals so I suppose that ad error can occur anytime (even during content playback). Calling endLinearAdMode works only when contrib-ads is inAdBreak(). When ad is loading (adtimeout is running): inAdBreak() === false as you can see in first post. I tested it with Google Ima tags Redirect Error sample.

I think that isAdLoading method is very important if you want let integration to fully handle ad errors.

Btw. isContentResuming is already included (doc). 

Thanks.

**Comment by incompl** [comment]
OK makes sense. I'm not able to work on it today but I think the implementation will follow how it's done for isContentResuming.

**Comment by mysuf** [comment]
Ok. I did it and tested it on few tags. Seems working. Dunno if it respects your coding style (your husky bit me few times) so I will make PR only on your command.

**Comment by incompl** [comment]
Please do open a PR, I'm happy to review and suggest any necessary changes.

**Comment by incompl** [comment]
Implemented as `isWaitingForAdBreak` in 6.2.0.
